### DISCUSSION TITLE
Tick Generator in MIDI

### DISCUSSION THREAD

**Discussion by Fazik74** [naudio/NAudio#1086] [Q&A]
Hi
I can't see what is a midi tick generator in MIDI module? Example: PPQ=480, this is about 1ms / tick. What generates such impulses in NAudio>Midi? Where is it ?

Thank you very much
Best regards,
Thomas

**Comment by markheath** [comment]
hi, I'm afraid NAudio does not offer this capability. You might want to check out MeltySynth, which integrates nicely with NAudio for support for sequencing MIDI. https://github.com/sinshu/meltysynth

**Reply by Fazik74** [reply]
Thank you Mark !
I created a tick generators for sequencer:C++, C#, unfortunately operating with microseconds is very CPU intensive.
I wonder HOW simple midi players work? Old programs like XG Works: CPU 0%. Atari 1040, Amiga with PPQN 480? No problem !
Do you have any suggestions?
Best regards,
Thomas
